src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -13169,6 +13239,12 @@ private bool IsInAsync
             }
         }
 
+        private bool ForceConditionalAccessExpression
+        {
+            get => _syntaxFactoryContext.ForceConditionalAccessExpression;
+            set => _syntaxFactoryContext.ForceConditionalAccessExpression = value;
+        }
```

Comment:
note: i went this route (surrounding context) because trying to actually use parameters to pass in this context inwards was extremely complex and viral.  Virtually all paths in expression parsing needed to pass this information along.  It both clutered things enormously, and made me worried about extra stack frame sizes.

The current approach is nice because it allows each level of ternary parsing to effectively retry the lower level with the old parsing, preferring that if it keeps the original parsing semantics.